src/main/drivers/bus_i2c_stm32f10x.c

```diff
@@ -41,56 +43,79 @@ static void i2cUnstick(IO_t scl, IO_t sda);
 
 #ifdef STM32F4
 
-#if defined(USE_I2C_PULLUP)
-#define IOCFG_I2C IO_CONFIG(GPIO_Mode_AF, 0, GPIO_OType_OD, GPIO_PuPd_UP)
+# if defined(USE_I2C_PULLUP)
+#  define IOCFG_I2C    IO_CONFIG(GPIO_Mode_AF, 0, GPIO_OType_OD, GPIO_PuPd_UP)
+# else
+#  define IOCFG_I2C    IOCFG_AF_OD
+# endif
+
+# define DEF_I2C1_SCL PB8
+# define DEF_I2C1_SDA PB9
+
 #else
-#define IOCFG_I2C IOCFG_AF_OD
-#endif
 
-#ifndef I2C1_SCL
-#define I2C1_SCL PB8
-#endif
-#ifndef I2C1_SDA
-#define I2C1_SDA PB9
+# define IOCFG_I2C    IO_CONFIG(GPIO_Mode_AF_OD, GPIO_Speed_50MHz)
+# define DEF_I2C1_SCL PB6
+# define DEF_I2C1_SDA PB7
+
 #endif
 
-#else
+#define DEF_I2C2_SCL  PB10
+#define DEF_I2C2_SDA  PB11
+
+#ifdef STM32F4
+# define DEF_I2C3_SCL PA8
+# define DEF_I2C3_SDA PC9
+#endif
 
 #ifndef I2C1_SCL
-#define I2C1_SCL PB6
+# define I2C1_SCL     NONE
 #endif
 #ifndef I2C1_SDA
-#define I2C1_SDA PB7
-#endif
-#define IOCFG_I2C   IO_CONFIG(GPIO_Mode_AF_OD, GPIO_Speed_50MHz)
-
+# define I2C1_SDA     NONE
 #endif
 
 #ifndef I2C2_SCL
-#define I2C2_SCL PB10
+# define I2C2_SCL     NONE
 #endif
-
 #ifndef I2C2_SDA
-#define I2C2_SDA PB11
+# define I2C2_SDA     NONE
 #endif
 
-#ifdef STM32F4
 #ifndef I2C3_SCL
-#define I2C3_SCL PA8
+# define I2C3_SCL     NONE
 #endif
 #ifndef I2C3_SDA
-#define I2C3_SDA PC9
-#endif
+# define I2C3_SDA     NONE
 #endif
 
-static i2cDevice_t i2cHardwareMap[] = {
-    { .dev = I2C1, .scl = IO_TAG(I2C1_SCL), .sda = IO_TAG(I2C1_SDA), .rcc = RCC_APB1(I2C1), .overClock = I2C1_OVERCLOCK, .ev_irq = I2C1_EV_IRQn, .er_irq = I2C1_ER_IRQn },
-    { .dev = I2C2, .scl = IO_TAG(I2C2_SCL), .sda = IO_TAG(I2C2_SDA), .rcc = RCC_APB1(I2C2), .overClock = I2C2_OVERCLOCK, .ev_irq = I2C2_EV_IRQn, .er_irq = I2C2_ER_IRQn },
+// List of possible I2C mapping (exported to bus_i2c.c)
+i2cDevice_t i2cHardwareMap[] = {
```

Comment:
Eventually something like https://github.com/betaflight/betaflight/blob/master/src/main/drivers/timer_def.h may be used to infer common values.. 

Also it may be better to separate F1 and F4 declarations (and removing `DEF_` indirection)